Index: third-party-src/extensions/shared.webpack.config.mjs
===================================================================
--- third-party-src.orig/extensions/shared.webpack.config.mjs
+++ third-party-src/extensions/shared.webpack.config.mjs
@@ -17,6 +17,7 @@ const require = createRequire(import.met
 const tsLoaderOptions = {
 	compilerOptions: {
 		'sourceMap': true,
+		'rootDir': '.',
 	},
 	onlyCompileBundledFiles: true,
 };
