Index: third-party-src/build/lib/getVersion.ts
===================================================================
--- third-party-src.orig/build/lib/getVersion.ts
+++ third-party-src/build/lib/getVersion.ts
@@ -12,5 +12,5 @@ export function getVersion(root: string)
 		version = git.getVersion(root);
 	}
 
-	return version;
+	return version || "unknown";
 }
Index: third-party-src/build/gulpfile.reh.ts
===================================================================
--- third-party-src.orig/build/gulpfile.reh.ts
+++ third-party-src/build/gulpfile.reh.ts
@@ -33,6 +33,7 @@ import log from 'fancy-log';
 import buildfile from './buildfile.ts';
 import { fetchUrls, fetchGithub } from './lib/fetch.ts';
 import jsonEditor from 'gulp-json-editor';
+import through from 'through2';
 
 
 const REPO_ROOT = path.dirname(import.meta.dirname);
@@ -197,6 +198,15 @@ function nodejs(platform: string, arch:
 		arch = 'x64';
 	}
 
+	if (process.env['CUSTOM_NODE_PATH']) {
+		log(`Bundling custom Node from '${process.env['CUSTOM_NODE_PATH']}'...`);
+		return gulp.src(process.env['CUSTOM_NODE_PATH'])
+			.pipe(flatmap(stream => stream.pipe(gunzip()).pipe(untar())))
+			.pipe(filter('**/node'))
+			.pipe(util.setExecutableBit('**'))
+			.pipe(rename('node'));
+	}
+
 	log(`Downloading node.js ${nodeVersion} ${platform} ${arch} from ${product.nodejsRepository}...`);
 
 	const glibcPrefix = process.env['VSCODE_NODE_GLIBC'] ?? '';
@@ -251,7 +261,7 @@ function nodejs(platform: string, arch:
 	}
 }
 
-function packageTask(type: string, platform: string, arch: string, sourceFolderName: string, destinationFolderName: string) {
+function packageTask(type: string, platform: string, arch: string, sourceFolderName: string, destinationFolderName: string, isSlim = false) {
 	const destination = path.join(BUILD_ROOT, destinationFolderName);
 
 	return () => {
@@ -328,6 +338,7 @@ function packageTask(type: string, platf
 			}));
 
 		const license = gulp.src(['remote/LICENSE'], { base: 'remote', allowEmpty: true });
+		const thirdPartylicense = gulp.src(['LICENSE-THIRD-PARTY'], { base: '.', allowEmpty: true });
 
 		const jsFilter = util.filter(data => !data.isDirectory() && /\.js$/.test(data.path));
 
@@ -343,7 +354,10 @@ function packageTask(type: string, platf
 			.pipe(jsFilter.restore);
 
 		const nodePath = `.build/node/v${nodeVersion}/${platform}-${arch}`;
-		const node = gulp.src(`${nodePath}/**`, { base: nodePath, dot: true });
+		// Do not include Node in slim package
+		const emptyTask = through.obj();
+		emptyTask.end();
+		const node = isSlim ? emptyTask : gulp.src(`${nodePath}/**`, { base: nodePath, dot: true });
 
 		let web: NodeJS.ReadWriteStream[] = [];
 		if (type === 'reh-web') {
@@ -359,6 +373,7 @@ function packageTask(type: string, platf
 			packageJsonStream,
 			productJsonStream,
 			license,
+			thirdPartylicense,
 			sources,
 			deps,
 			node,
@@ -385,20 +400,24 @@ function packageTask(type: string, platf
 					.pipe(rename(`bin/${product.serverApplicationName}.cmd`)),
 			);
 		} else if (platform === 'linux' || platform === 'alpine' || platform === 'darwin') {
+			const nodePathStr = isSlim ? 'node' : '"$ROOT/node"';
 			result = es.merge(result,
 				gulp.src(`resources/server/bin/remote-cli/${platform === 'darwin' ? 'code-darwin.sh' : 'code-linux.sh'}`, { base: '.' })
 					.pipe(replace('@@VERSION@@', version))
 					.pipe(replace('@@COMMIT@@', commit || ''))
 					.pipe(replace('@@APPNAME@@', product.applicationName))
+					.pipe(replace('"$ROOT/node"', nodePathStr))
 					.pipe(rename(`bin/remote-cli/${product.applicationName}`))
 					.pipe(util.setExecutableBit()),
 				gulp.src(`resources/server/bin/helpers/${platform === 'darwin' ? 'browser-darwin.sh' : 'browser-linux.sh'}`, { base: '.' })
 					.pipe(replace('@@VERSION@@', version))
 					.pipe(replace('@@COMMIT@@', commit || ''))
 					.pipe(replace('@@APPNAME@@', product.applicationName))
+					.pipe(replace('"$ROOT/node"', nodePathStr))
 					.pipe(rename(`bin/helpers/browser.sh`))
 					.pipe(util.setExecutableBit()),
 				gulp.src(`resources/server/bin/${platform === 'darwin' ? 'code-server-darwin.sh' : 'code-server-linux.sh'}`, { base: '.' })
+					.pipe(replace('"$ROOT/node"', nodePathStr))
 					.pipe(rename(`bin/${product.serverApplicationName}`))
 					.pipe(util.setExecutableBit())
 			);
@@ -465,6 +484,13 @@ function tweakProductForServerWeb(produc
 		['', 'min'].forEach(minified => {
 			const sourceFolderName = `out-vscode-${type}${dashed(minified)}`;
 			const destinationFolderName = `vscode-${type}${dashed(platform)}${dashed(arch)}`;
+			const slimDestinationFolderName = `${destinationFolderName}-slim`;
+
+			const slimServerTaskCI = task.define(`vscode-${type}${dashed(platform)}${dashed(arch)}${dashed(minified)}-ci-slim`, task.series(
+				util.rimraf(path.join(BUILD_ROOT, slimDestinationFolderName)),
+				packageTask(type, platform, arch, sourceFolderName, slimDestinationFolderName, true)
+			));
+			gulp.task(slimServerTaskCI);
 
 			const serverTaskCI = task.define(`vscode-${type}${dashed(platform)}${dashed(arch)}${dashed(minified)}-ci`, task.series(
 				compileNativeExtensionsBuildTask,
