Index: third-party-src/src/vs/workbench/browser/parts/globalCompositeBar.ts
===================================================================
--- third-party-src.orig/src/vs/workbench/browser/parts/globalCompositeBar.ts
+++ third-party-src/src/vs/workbench/browser/parts/globalCompositeBar.ts
@@ -132,7 +132,7 @@ export class GlobalCompositeBar extends
 	}
 
 	getContextMenuActions(): IAction[] {
-		return [toAction({ id: 'toggleAccountsVisibility', label: localize('accounts', "Accounts"), checked: this.accountsVisibilityPreference, run: () => this.accountsVisibilityPreference = !this.accountsVisibilityPreference })];
+		return [];
 	}
 
 	private toggleAccountsActivity() {
@@ -147,7 +147,7 @@ export class GlobalCompositeBar extends
 	}
 
 	private get accountsVisibilityPreference(): boolean {
-		return isAccountsActionVisible(this.storageService);
+		return false;
 	}
 
 	private set accountsVisibilityPreference(value: boolean) {
Index: third-party-src/src/vs/workbench/contrib/authentication/browser/authentication.contribution.ts
===================================================================
--- third-party-src.orig/src/vs/workbench/contrib/authentication/browser/authentication.contribution.ts
+++ third-party-src/src/vs/workbench/contrib/authentication/browser/authentication.contribution.ts
@@ -88,13 +88,6 @@ class AuthenticationContribution extends
 	}
 
 	private _registerActions(): void {
-		this._register(registerAction2(ManageAccountsAction));
-		this._register(registerAction2(SignOutOfAccountAction));
-		this._register(registerAction2(ManageTrustedExtensionsForAccountAction));
-		this._register(registerAction2(ManageAccountPreferencesForExtensionAction));
-		this._register(registerAction2(ManageTrustedMcpServersForAccountAction));
-		this._register(registerAction2(ManageAccountPreferencesForMcpServerAction));
-		this._register(registerAction2(RemoveDynamicAuthenticationProvidersAction));
 	}
 }
 
@@ -210,7 +203,7 @@ class AuthenticationUsageContribution im
 // 	}
 // }
 
-registerWorkbenchContribution2(AuthenticationContribution.ID, AuthenticationContribution, WorkbenchPhase.AfterRestored);
-registerWorkbenchContribution2(AuthenticationUsageContribution.ID, AuthenticationUsageContribution, WorkbenchPhase.Eventually);
+// registerWorkbenchContribution2(AuthenticationContribution.ID, AuthenticationContribution, WorkbenchPhase.AfterRestored);
+// registerWorkbenchContribution2(AuthenticationUsageContribution.ID, AuthenticationUsageContribution, WorkbenchPhase.Eventually);
 // registerWorkbenchContribution2(AuthenticationExtensionsContribution.ID, AuthenticationExtensionsContribution, WorkbenchPhase.Eventually);
 // registerWorkbenchContribution2(AuthenticationMcpContribution.ID, AuthenticationMcpContribution, WorkbenchPhase.Eventually);
