Index: third-party-src/src/vs/workbench/browser/actions/helpActions.ts
===================================================================
--- third-party-src.orig/src/vs/workbench/browser/actions/helpActions.ts
+++ third-party-src/src/vs/workbench/browser/actions/helpActions.ts
@@ -16,7 +16,6 @@ import { ServicesAccessor } from '../../../platform/instantiation/common/instant
 import { KeybindingWeight } from '../../../platform/keybinding/common/keybindingsRegistry.js';
 import { Categories } from '../../../platform/action/common/actionCommonCategories.js';
 import { ICommandService } from '../../../platform/commands/common/commands.js';
-import { ContextKeyExpr } from '../../../platform/contextkey/common/contextkey.js';
 
 class KeybindingsReferenceAction extends Action2 {
 
@@ -332,35 +331,6 @@ class GetStartedWithAccessibilityFeatures extends Action2 {
 	}
 }
 
-class AskVSCodeCopilot extends Action2 {
-	static readonly ID = 'workbench.action.askVScode';
-
-	constructor() {
-		super({
-			id: AskVSCodeCopilot.ID,
-			title: localize2('askVScode', 'Ask @vscode'),
-			category: Categories.Help,
-			f1: true,
-			precondition: ContextKeyExpr.equals('chatSetupHidden', false)
-		});
-	}
-
-	async run(accessor: ServicesAccessor): Promise<void> {
-		const commandService = accessor.get(ICommandService);
-		commandService.executeCommand('workbench.action.chat.open', { mode: 'ask', query: '@vscode ', isPartialQuery: true });
-	}
-}
-
-MenuRegistry.appendMenuItem(MenuId.MenubarHelpMenu, {
-	command: {
-		id: AskVSCodeCopilot.ID,
-		title: localize2('askVScode', 'Ask @vscode'),
-	},
-	order: 7,
-	group: '1_welcome',
-	when: ContextKeyExpr.equals('chatSetupHidden', false)
-});
-
 // --- Actions Registration
 
 if (KeybindingsReferenceAction.AVAILABLE) {
@@ -400,5 +370,3 @@ if (OpenPrivacyStatementUrlAction.AVAILABLE) {
 }
 
 registerAction2(GetStartedWithAccessibilityFeatures);
-
-registerAction2(AskVSCodeCopilot);
