Index: third-party-src/src/vs/server/node/serverEnvironmentService.ts
===================================================================
--- third-party-src.orig/src/vs/server/node/serverEnvironmentService.ts
+++ third-party-src/src/vs/server/node/serverEnvironmentService.ts
@@ -9,6 +9,7 @@ import { NativeEnvironmentService } from
 import { OPTIONS, OptionDescriptions } from '../../platform/environment/node/argv.js';
 import { refineServiceDecorator } from '../../platform/instantiation/common/instantiation.js';
 import { IEnvironmentService, INativeEnvironmentService } from '../../platform/environment/common/environment.js';
+import { IProductService } from '../../platform/product/common/productService.js';
 import { memoize } from '../../base/common/decorators.js';
 import { URI } from '../../base/common/uri.js';
 import { joinPath } from '../../base/common/resources.js';
@@ -247,6 +248,16 @@ export interface IServerEnvironmentServi
 }
 
 export class ServerEnvironmentService extends NativeEnvironmentService implements IServerEnvironmentService {
+
+	constructor(args: ServerParsedArgs, productService: IProductService) {
+		/* ----- sagemaker compatibility: map --base-path to --server-base-path ----- */
+		if (args['base-path']) {
+			args['server-base-path'] = args['base-path'];
+		}
+
+		super(args, productService);
+	}
+
 	@memoize
 	override get userRoamingDataHome(): URI { return this.appSettingsHome; }
 	@memoize
