Index: third-party-src/build/gulpfile.vscode.web.ts
===================================================================
--- third-party-src.orig/build/gulpfile.vscode.web.ts
+++ third-party-src/build/gulpfile.vscode.web.ts
@@ -82,6 +82,7 @@ const vscodeWebEntryPoints = [
 	buildfile.workerBackgroundTokenization,
 	buildfile.keyboardMaps,
 	buildfile.workbenchWeb,
+	buildfile.codeWeb,
 	buildfile.entrypoint('vs/workbench/workbench.web.main.internal') // TODO@esm remove line when we stop supporting web-amd-esm-bridge
 ].flat();
 
