Index: third-party-src/src/vs/workbench/contrib/chat/browser/chat.contribution.ts
===================================================================
--- third-party-src.orig/src/vs/workbench/contrib/chat/browser/chat.contribution.ts
+++ third-party-src/src/vs/workbench/contrib/chat/browser/chat.contribution.ts
@@ -1198,26 +1198,6 @@ registerWorkbenchContribution2(UserToolS
 registerWorkbenchContribution2(PromptLanguageFeaturesProvider.ID, PromptLanguageFeaturesProvider, WorkbenchPhase.Eventually);
 registerWorkbenchContribution2(ChatWindowNotifier.ID, ChatWindowNotifier, WorkbenchPhase.AfterRestored);
 
-registerChatActions();
-registerChatAccessibilityActions();
-registerChatCopyActions();
-registerChatCodeBlockActions();
-registerChatCodeCompareBlockActions();
-registerChatFileTreeActions();
-registerChatPromptNavigationActions();
-registerChatTitleActions();
-registerChatExecuteActions();
-registerQuickChatActions();
-registerChatExportActions();
-registerMoveActions();
-registerNewChatActions();
-registerChatContextActions();
-registerChatDeveloperActions();
-registerChatEditorActions();
-registerChatElicitationActions();
-registerChatToolActions();
-registerLanguageModelActions();
-registerAction2(ConfigureToolSets);
 registerEditorFeature(ChatPasteProvidersFeature);
 
 
