Index: third-party-src/build/gulpfile.vscode.web.ts
===================================================================
--- third-party-src.orig/build/gulpfile.vscode.web.ts
+++ third-party-src/build/gulpfile.vscode.web.ts
@@ -56,7 +56,10 @@ export const vscodeWebResourceIncludes =
 	'out-build/vs/editor/common/languages/injections/*.scm',
 
 	// Extension Host Worker
-	'out-build/vs/workbench/services/extensions/worker/webWorkerExtensionHostIframe.html'
+	'out-build/vs/workbench/services/extensions/worker/webWorkerExtensionHostIframe.html',
+
+	// Error handler
+	'out-build/vs/editor/common/errors/suppressedErrors.js'
 ];
 
 const vscodeWebResources = [
